api/api-archive-search/archive-search-internal/src/main/java/fr/gouv/vitamui/archive/internal/server/service/ArchivesSearchManagementRulesQueryBuilderService.java

```diff
@@ -163,6 +174,18 @@ private void buildRuleIdentifierQuery(final List<String> searchValues,
             subQueryAnd.add(subQueryOr);
         }
     }
+    private void buildInheritedCategoryQuery(final List<String> searchValues,
+        ArchiveSearchConsts.CriteriaOperators operator, BooleanQuery subQueryAnd)
+        throws InvalidCreateOperationException {
+        BooleanQuery subQueryOr = or();
+        if (!CollectionUtils.isEmpty(searchValues)) {
+            for (String value : searchValues) {
+                subQueryOr
+                    .add(VitamQueryHelper.buildSubQueryByOperator(ArchiveSearchConsts.APPRAISAL_RULE_INHERITED, value, operator));
+            }
+            subQueryAnd.add(subQueryOr);
+        }
+    }
```

Comment:
ppour la lisibilité, je préfère ne pas imbriquer les if else
et pour l'iteration un foreach genre: 
```
       if (CollectionUtils.isEmpty(searchValues)) {
            return;
        }
        BooleanQuery subQueryOr = or();
        searchValues.forEach(value-> subQueryOr
            .add(VitamQueryHelper.buildSubQueryByOperator(ArchiveSearchConsts.APPRAISAL_RULE_INHERITED, value, operator)));
        subQueryAnd.add(subQueryOr);
```

Reply:
why not, bloc modifié